prometheus/src/main/scala/filodb/prometheus/ast/Functions.scala

```diff
@@ -97,20 +99,25 @@ trait Functions extends Base with Operators with Vectors {
         val periodicSeriesPlan = seriesParam.asInstanceOf[PeriodicSeries].toPeriodicSeriesPlan(timeParams)
         val params = RangeParams(timeParams.start, timeParams.step, timeParams.end)
         ScalarVaryingDoublePlan(periodicSeriesPlan, scalarFunctionIdOpt.get, params)
-      }
-      else if (sortFunctionIdOpt.isDefined) {
+      } else if (sortFunctionIdOpt.isDefined) {
         val sortFunctionId = sortFunctionIdOpt.get
         val periodicSeriesPlan = seriesParam.asInstanceOf[PeriodicSeries].toPeriodicSeriesPlan(timeParams)
         ApplySortFunction(periodicSeriesPlan, sortFunctionId)
-      } else {
-        val rangeFunctionId = RangeFunctionId.withNameInsensitiveOption(name).get
-        val rangeExpression = seriesParam.asInstanceOf[RangeExpression]
-        PeriodicSeriesWithWindowing(
-          rangeExpression.toRawSeriesPlan(timeParams, isRoot = false).asInstanceOf[RawSeries],
-          timeParams.start * 1000, timeParams.step * 1000, timeParams.end * 1000,
-          rangeExpression.window.millis,
-          rangeFunctionId, otherParams)
+      } else if (absentFunctionIdOpt.isDefined) {
+        val columnFilter = seriesParam.asInstanceOf[InstantExpression].columnFilters
+        val periodicSeriesPlan = seriesParam.asInstanceOf[PeriodicSeries].toPeriodicSeriesPlan(timeParams)
+        ApplyAbsentFunction(periodicSeriesPlan, columnFilter, RangeParams(timeParams.start, timeParams.step,
+          timeParams.end))
+        } else {
```

Comment:
There is something weird with the indentation here.

Reply:
Fixed it